src/solver/pad_reflection/pad_reflection.cpp

```diff
@@ -0,0 +1,158 @@
+{
+    if(!problem.IsSameType())
+        return false;
+    if(!problem.IsAllPacked())
+        return false;
+    if(!problem.IsRightNumPadding())
+        return false;
+    return true;
+}
+
+ConvSolution
+PadReflection::GetSolution([[maybe_unused]] const ExecutionContext& context,
+                           const miopen::pad_reflection::ProblemDescription& problem) const
+{
+    auto result = ConvSolution{miopenStatusSuccess};
+
+    auto input_dtype  = miopen::GetDataType(problem.GetXDesc().GetType());
+    auto output_dtype = miopen::GetDataType(problem.GetYDesc().GetType());
+    auto xdims        = problem.GetXDesc().GetLengths();
+    auto ydims        = problem.GetYDesc().GetLengths();
+
+    {
+        auto kernel = KernelInfo{};
+
+        kernel.kernel_file = "MIOpenPadReflection.cpp";
+        kernel.kernel_name = "PadReflection2dFwdContiguous";
+        auto output_numel =
+            std::accumulate(ydims.begin(), ydims.end(), 1ULL, std::multiplies<size_t>());
+
+        const auto build_params = KernelBuildParameters{
+            {"INPUT_TYPE", input_dtype == "bfloat16" ? "ushort" : input_dtype},
+            {"OUTPUT_TYPE", output_dtype == "bfloat16" ? "ushort" : output_dtype},
+        };
+
+        kernel.comp_options = build_params.GenerateFor(kbp::HIP{});
+
+        size_t xlocalsize = LOCAL_SIZE;
+        size_t xgridsize  = AlignUp(output_numel, xlocalsize);
+        size_t ylocalsize = 1;
+        size_t ygridsize  = 1;
+        size_t zlocalsize = 1;
+        size_t zgridsize  = 1;
+        kernel.l_wk.push_back(xlocalsize);
+        kernel.l_wk.push_back(ylocalsize);
+        kernel.l_wk.push_back(zlocalsize);
+
+        kernel.g_wk.push_back(xgridsize);
+        kernel.g_wk.push_back(ygridsize);
+        kernel.g_wk.push_back(zgridsize);
+
+        result.construction_params.push_back(kernel);
+    }
+    result.invoker_factory = [](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle_, const AnyInvokeParams& raw_params) {
+            decltype(auto) kernel = handle_.Run(kernels[0]);
+            decltype(auto) params = raw_params.CastTo<miopen::pad_reflection::InvokeParams>();
+
+            auto xdims = params.xDesc->GetLengths();
+            auto ydims = params.yDesc->GetLengths();
+
+            auto xstrides = params.xDesc->GetStrides();
+
+            auto output_size =
+                std::accumulate(ydims.begin(), ydims.end(), 1ULL, std::multiplies<size_t>());
+
+            auto padding     = params.padding;
+            auto num_padding = params.num_padding;
+            long padding_l, padding_t;
+            if(num_padding == 1)
+            {
+                padding_l = padding[0];
+                padding_t = padding[0];
+            }
+            else if(num_padding == 4)
+            {
+                padding_l = padding[0];
+                padding_t = padding[2];
+            }
+            // long padding_l        = (*padding)[0];
```

Comment:
Remove unused lines.

Reply:
resolved in https://github.com/moreh-dev/MIOpen/pull/7/commits/a008c17039ef89a729d072c0f174df1312b932f9